crates/net/network/src/config.rs

```diff
@@ -83,6 +84,8 @@ pub struct NetworkConfig<C, N: NetworkPrimitives = EthNetworkPrimitives> {
     pub transactions_manager_config: TransactionsManagerConfig,
     /// The NAT resolver for external IP
     pub nat: Option<NatResolver>,
+    /// The P2P handshake
+    pub handshake: Arc<dyn Handshake>,
```

Comment:
I'd like to name this `EthRlpxHandshake` and add a few more docs here why we need this